### PR TITLE
Change flatpak module to use the --noninteractive CLI argument

### PR DESCRIPTION
**Pull request by jdufresne**
##### SUMMARY

The --noninteractive argument is documented as:

    $ flatpak install --help
      …
      --noninteractive        Produce minimal output and don't ask questions

This removes the poorly formatted output of the commands due to the use
of non-serial output. For example, with the yaml output plugin, Ansible
previously formatted the flatpak command output as (notice the
repetitive "Installing" on one line):

    TASK [flatpak : Install flatpak applications]
    changed: [localhost] => (item=com.slack.Slack) => changed=true
      ansible_loop_var: item
      command: /usr/bin/flatpak install --user -y flathub com.slack.Slack
      item: com.slack.Slack
      rc: 0
      stderr: ''
      stderr_lines: <omitted>
      stdout: |-
        Looking for matches

        com.slack.Slack permissions:
            ipc network pulseaudio      x11
            devices     file access [1] dbus access [2]

            [1] xdg-documents:ro, xdg-download, xdg-music:ro, xdg-pictures:ro,
                xdg-run/dconf, xdg-videos:ro, ~/.config/dconf:ro
            [2] ca.desrt.dconf, org.freedesktop.Notifications,
                org.kde.StatusNotifierWatcher

         1.             com.slack.Slack x86_64  stable  flathub < 62.4MB

        Installing                                                                   Installing                       1%Installing                      9%Installing                      9%  880.8kB/sInstalling                      9%  1.1MB/sInstalling                    14%  2.2MB/sInstalling                  21%  3.0MB/sInstalling                 27%  3.3MB/s  00:10Installing                33%  3.4MB/s  00:10Installing               39%  3.6MB/s  00:09Installing             46%  3.7MB/s  00:08Installing            52%  3.8MB/s  00:07Installing           58%  3.8MB/s  00:06Installing          65%  3.9MB/s  00:05Installing        71%  3.9MB/s  00:04Installing       78%  4.0MB/s  00:03Installing      84%  4.0MB/s  00:02Installing    91%  4.0MB/s  00:01Installing   97%  4.0MB/s  00:00Installing  100%  4.2MB/s  00:00Installing  100%  3.7MB/s  00:00Installation complete.
      stdout_lines: <omitted>

Now, the output is formatted simply as as:

    Installing app/com.slack.Slack/x86_64/stable

The use of the --noninteractive argument obviates the need to use the -y
argument.

The --noninteractive argument was added in flatpak 1.1.3.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
flatpak

### CODE REVIEW

File: changelogs/fragments/268-flatpak-noninteractive-output.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+- flatpak - fix output by running with the --noninteractive argument.
```
Comment (Andersson007) [inline changelogs/fragments/268-flatpak-noninteractive-output.yaml:2]:
```suggestion
- flatpak - fix output by running with the --noninteractive argument (https://github.com/ansible-collections/community.general/pull/268).
```

Comment (jdufresne) [reply changelogs/fragments/268-flatpak-noninteractive-output.yaml]: Thanks. Applied.

**Comment by Andersson007** [comment]
The option `noninteractive` was released in Jan 2019.
I think the changes can break current users pipelines who are using older versions of flatpak.

**Comment by jdufresne** [comment]
True. Is there a policy on flatpak version support or flatpak environments targeted?

The current output is pretty awful in logs.

**Comment by felixfontein** [comment]
Not really. A module can require whatever it wants. It's usually nice though not to unnecessarily break backwards compatibililty.

What you could do is run something like `flatpak --version` (I don't have flatpak on this machine, so no idea what the correct option is) and see whether it is new enough, and only use `noninteractive` in that case. Then it will continue to work with older versions, and look better with newer ones.

(The docker connection plugin does something like that, for example: https://github.com/ansible-collections/community.general/blob/master/plugins/connection/docker.py#L89-L93)

**Comment by Andersson007** [comment]
maybe catching the exception (+ check the exception msg like `noninteractive option is not supported` or something) should also work

**Comment by Andersson007** [comment]
any updates on this?

**Comment by jdufresne** [comment]
Sorry, I have not had time to resume work on this. If someone else would like to complete the work and take it over the finish line, that would be welcome. Cheers.

**Comment by TGuimbert** [comment]
Hi, I would like to work on this. How should I proceed, create a new PR from a fork ?

**Comment by felixfontein** [comment]
@TGuimbert hmm good question. I guess it depends on your git fu a bit :) The easiest is to clone this repository and then check out this PR (see for example https://stackoverflow.com/questions/27567846/how-can-i-check-out-a-github-pull-request-with-git). Then you need to rebase the new branch against `main` to resolve the conflicts.

**Comment by felixfontein** [comment]
You probably have to fork the repo first (if you haven't done that already), add this repo as a remote, and then check out this PR from this repo. It might be a bit more tricky than checkout out the PR in a clone of this repo directly. And then you need to push your branch back to your fork, and create a PR from it.

**Comment by jdufresne** [comment]
Closing in favor of #1246.

**Pull request closed without merging**